rita/src/rita_common/network_endpoints/mod.rs

```diff
@@ -79,17 +80,22 @@ pub fn hello_response(
     // We send the callback, which can safely allocate a port because it already successfully
     // contacted a neighbor. The exception to this is when the TCP session fails at exactly
     // the wrong time.
-    TunnelManager::from_registry()
-        .send(IdentityCallback::new(their_id, peer, None))
-        .and_then(|tunnel| {
-            let tunnel = tunnel.unwrap();
-            Ok(Json(LocalIdentity {
-                global: SETTING.get_identity(),
-                wg_port: tunnel.0.listen_port,
-                have_tunnel: Some(tunnel.1),
-            }))
-        }).from_err()
-        .responder()
+    Box::new(
+        TunnelManager::from_registry()
+            .send(IdentityCallback::new(their_id, peer, None))
```

Comment:
move the from_err() up here right after the send. The issue is that you have a mailbox error (the sort of thing you get from a failed send) and you need to  convert it to a failure error to be consistent with what you return on line 91